core/plugins/org.fusesource.ide.camel.model.service.core/src/org/fusesource/ide/camel/model/service/core/util/CamelComponentUtils.java

```diff
@@ -199,102 +205,111 @@ public static boolean isUnsupportedProperty(Parameter p) {
 		return res.toArray(new String[res.size()]);
 	}
 
-	/**
-	 * returns the component class for the given scheme
-	 *
-	 * @param scheme
-	 * @return the class or null if not found
-	 */
-	protected static String getComponentClass(String scheme, CamelFile camelFile) {
+	public static String getComponentClassForScheme(String scheme, CamelFile camelFile) {
 		String compClass = null;
-		IProject project = camelFile.getResource().getProject();
 		List<Component> components = camelFile.getCamelModel().getComponentModel().getSupportedComponents();
 		for (Component c : components) {
 			if (c.supportsScheme(scheme)) {
 				compClass = c.getClazz();
 				break;
 			}
 		}
+		return compClass;
+	}
 
-		if (compClass == null) {
-			// seems this scheme has no model entry -> check dependency
-			try {
-				ZipFile zf = null;
-				ZipEntry ze = null;
-				IJavaProject jpr = JavaCore.create(project);
-				for (IClasspathEntry e : jpr.getResolvedClasspath(true)) {
-					File cpEntryFile = e.getPath().toFile();
-					if (isJarFile(cpEntryFile)) {
-						zf = new ZipFile(cpEntryFile);
-						ze = zf.getEntry(String.format("META-INF/services/org/apache/camel/component/%s", scheme));
-						if (ze != null) {
-							break;
-						}
-						ze = null;
-						zf = null;
-					}
-				}
-
+	private static String getClassFromZipEntry(String scheme, File cpEntryFile) {
+		if (isJarFile(cpEntryFile)) {
+			try (ZipFile zf = new ZipFile(cpEntryFile)) {
+				ZipEntry ze = zf.getEntry(String.format("META-INF/services/org/apache/camel/component/%s", scheme));
 				if (ze != null) {
 					// try to generate a model entry for the component class
 					Properties p = new Properties();
 					p.load(zf.getInputStream(ze));
-					compClass = p.getProperty("class");
-				}
-			} catch (Exception ex) {
+					return p.getProperty("class");
+				} 
+			} catch (IOException ex) {
 				CamelModelServiceCoreActivator.pluginLog().logError(ex);
-				compClass = null;
 			}
 		}
-
-		return compClass;
+		return null;
 	}
-
+	
+	private static String getComponentClassFromJar(IJavaProject jpr, String scheme){
+		try {
```

Comment:
What do you think to put the Try inside the loop?
like that if we have an invalid jars but which is not the one containing the component class we are interested in, it will still work.